pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/segmentgenerationandpush/SegmentGenerationAndPushTaskExecutor.java

```diff
@@ -113,6 +118,9 @@ public Object executeTask(PinotTaskConfig pinotTaskConfig)
     SegmentGenerationAndPushResult.Builder resultBuilder = new SegmentGenerationAndPushResult.Builder();
     File localTempDir = new File(new File(MinionContext.getInstance().getDataDir(), "SegmentGenerationAndPushResult"),
         "tmp-" + UUID.randomUUID());
+    _pinotTaskConfig = pinotTaskConfig;
+    _eventObserver = MinionEventObservers.getInstance().getMinionEventObserver(pinotTaskConfig.getTaskId());
+    _eventObserver.notifyProgress(pinotTaskConfig, "Task running");
```

Comment:
I feel that we may be able to define minion progress ENUM so that progress information is standardized?

Reply:
If standardization is wanted like by some automation, I'd suggest to go with the Helix task status for now. The status tracked via notifyProgress aims to be task specific to enrich the those coarse grained Helix task status.